### PR TITLE
KOTOR: Initial satellite camera implementation w\ PC movement

### PR DESCRIPTION
**Pull request by seedhartha**
Alters camera behavior in KotOR to match the original game. Press A/D to rotate the camera around PC, W/S to move PC forward or backwards. Use console command "tfc" to toggle free roam camera mode.

TODO: Use FOV, pitch and distance values from camerastyle.2da; smooth out camera rotation and PC movement

### CODE REVIEW

**Comment by DrMcCoy** [comment]
Could you describe a bit what this does and how, where, etc.? :)

**Comment by seedhartha** [comment]
Updated description.

File: src/engines/aurora/satellitecamera.cpp [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+	_target._z = z;
+}
+
+void SatelliteCamera::setDistance(float value) {
+	_distance = value;
+}
+
+void SatelliteCamera::setPitch(float value) {
+	_pitch = value;
+	auto pitchRad = Common::deg2rad(_pitch);
```
Comment (DrMcCoy) [inline src/engines/aurora/satellitecamera.cpp]: Don't use `auto`, please. We're still C++03 at the moment.

**Comment by DrMcCoy** [comment]
When toggling the free roam camera on, and moving the mouse while holding the middle mouse button (i.e. "free look"), it now lags quite a bit.

The issue is that you pulled the `CameraMan.update()` into the event queue loop in `Module::handleEvents()`. The update resorts all objects, so it is pretty taxing, and mouse move events fire in quick successions. My hacky resolution was to let the input handler process a whole batch before "finalizing" it by resorting the objects.

**Comment by seedhartha** [comment]
Fixed free roam camera lag. Also implemented idle and running animations for PC.

**Comment by DrMcCoy** [comment]
Very nice! Merged as b47c618b6cce739a8402dee0b99c9627a7814d37 (I word-wrapped the commit message to max 70 characters per line), thanks! :D

**Pull request closed without merging**